adalflow/adalflow/components/model_client/lm_studio_client.py

```diff
@@ -0,0 +1,120 @@
+
+    def __init__(self, host: Optional[str] = None, port: Optional[int] = None):
+        super().__init__()
+        self._host = host or os.getenv("LMSTUDIO_HOST", "http://localhost")
+        self._port = port or int(os.getenv("LMSTUDIO_PORT", "1234"))
+        self._base_url = f"{self._host}:{self._port}/v1"
+        self.init_sync_client()
+        self.async_client = None  # To be added
+
+    def init_sync_client(self):
+        """Create the synchronous client"""
+        self.sync_client = requests.Session()
+
+    def convert_inputs_to_api_kwargs(
+        self,
+        input: Optional[Any] = None,
+        model_kwargs: Dict = {},
+        model_type: ModelType = ModelType.UNDEFINED,
+    ) -> Dict:
+        """Convert the input and model_kwargs to api_kwargs for the LM Studio API."""
+        final_model_kwargs = model_kwargs.copy()
+        if model_type == ModelType.EMBEDDER:
+            if isinstance(input, str):
+                input = [input]
+            assert isinstance(input, Sequence), "input must be a sequence of text"
+            final_model_kwargs["input"] = input
+        elif model_type == ModelType.LLM:
+            messages = []
+            if input is not None and input != "":
+                messages.append({"role": "system", "content": "You are a helpful assistant. Provide a direct and concise answer to the user's question. Do not include any URLs or references in your response."})
+                messages.append({"role": "user", "content": input})
+            assert isinstance(messages, Sequence), "input must be a sequence of messages"
+            final_model_kwargs["messages"] = messages
+            
+            # Set default values for controlling response length if not provided
+            final_model_kwargs.setdefault("max_tokens", 50)
+            final_model_kwargs.setdefault("temperature", 0.1)
+            final_model_kwargs.setdefault("top_p", 0.9)
+            final_model_kwargs.setdefault("frequency_penalty", 0.0)
+            final_model_kwargs.setdefault("presence_penalty", 0.0)
+            final_model_kwargs.setdefault("stop", ["\n", "###", "://"])
+        else:
+            raise ValueError(f"model_type {model_type} is not supported")
+        return final_model_kwargs
+
+    @backoff.on_exception(backoff.expo, requests.RequestException, max_time=10)
+    def call(self, api_kwargs: Dict = {}, model_type: ModelType = ModelType.UNDEFINED):
+        if model_type == ModelType.EMBEDDER:
+            response = self.sync_client.post(f"{self._base_url}/embeddings", json=api_kwargs)
+        elif model_type == ModelType.LLM:
+            response = self.sync_client.post(f"{self._base_url}/chat/completions", json=api_kwargs)
+        else:
+            raise ValueError(f"model_type {model_type} is not supported")
+        
+        response.raise_for_status()
+        return response.json()
+
+    def parse_chat_completion(self, completion: Dict) -> GeneratorOutput:
```

Comment:
I suggest writting more precise error messages

```suggestion
def parse_chat_completion(self, completion: Dict) -> GeneratorOutput:
    """Parse the completion to a GeneratorOutput."""
    try:
        if "choices" not in completion:
            return GeneratorOutput(data=None, error="Error parsing the completion: 'choices' not in 'completion'.", raw_response=content) 

        elif not len(completion["choices"]) > 0:
            return GeneratorOutput(data=None, error="Error parsing the completion: 'choices' length is 0.", raw_response=content)

        else:
            content = completion["choices"][0]["message"]["content"]

            # Clean up the content
            content = self._clean_response(content)
            return GeneratorOutput(data=None, raw_response=content)
    except Exception as e:
        log.error(f"Error parsing the completion: {e}")
        return GeneratorOutput(data=None, error=str(e), raw_response=completion)
```